packages/ui/react/src/components/Primitives/ErrorMessage/ErrorMessage.tsx

```diff
@@ -0,0 +1,88 @@
+import { StyledErrorMessage } from './ErrorMessage.styled'
+
+/** Error with a type field that can be 'required' or any other string except 'required' */
+type ErrorWithType = { type: 'required' | Omit<string, 'required'> }
+
+/** Error containing a direct message string */
+type ErrorWithMessage = { message: string }
+
+/** API error response containing nested message */
+type APIError = { errors: { message: string } }
+
+type ErrorMessageProps = {
+  /** Renders the following error formats, by first match:
+   *
+   * 1. { type: 'required' }
+   *
+   * 2. { message: string  }
+   *
+   * 3. { errors: { message: string } } [1]
+   *
+   * [1] API Error response
+   */
+  error?: ErrorWithType | ErrorWithMessage | APIError | null
+
+  /** Additional CSS classes to apply to the error message container */
+  className?: string
+}
+
+/**
+ * Extracts an error message from various error object formats.
+ *
+ * @param error - Error object that can match one of these formats:
+ *                1. { type: 'required' }
+ *                2. { message: string }
+ *                3. { errors: { message: string } } - API Error response
+ * @returns The extracted error message string or null if no valid message is found
+ */
+function handleErrorMessage(error: ErrorMessageProps['error']) {
+  if (!error) return null
+
+  if ('type' in error) {
+    if (error.type === 'required') {
+      return 'This field is required!'
+    }
+  }
+
+  if ('message' in error) {
+    return error.message
+  }
+
+  if ('errors' in error) {
+    return error.errors.message
+  }
+
+  return null
+}
+
+/** Test ID for the error message component used in testing */
+const ERROR_MESSAGE_TEST_ID = 'error-message'
+
+/**
+ * Component that displays error messages in a standardized format.
+ * Handles multiple error object formats and renders them consistently.
+ *
+ * @param props - Component props
+ * @param props.error - Error object in various possible formats
+ * @param props.className - Additional CSS classes to apply to the container
+ */
+const ErrorMessage = ({ error, className }: ErrorMessageProps) => {
+  const errorMessage = handleErrorMessage(error)
+  const classNames = [
+    'translate',
+    className,
+  ]
+    .filter(Boolean)
+    .join(' ')
+
+  return (
+    <StyledErrorMessage
+      className={classNames}
+      data-testid={ERROR_MESSAGE_TEST_ID}
+    >
+      {errorMessage}
+    </StyledErrorMessage>
+  )
+}
+
```

Comment:
Can you apply the suggestion below? This helps keep a consistent pattern with other components by exporting the props as `<Component Name>Props`

```suggestion

export type { ErrorMessageProps }
```